CloudOnCloudImport/ShareViewController.swift

```diff
@@ -0,0 +1,154 @@
+        if itemProvider.hasItemConformingToTypeIdentifier(UTType.fileURL.identifier) {
+            saveData(itemProvider: itemProvider)
+        } else if itemProvider.hasItemConformingToTypeIdentifier(UTType.url.identifier) {
+            saveUrl(itemProvider: itemProvider)
+        } else if itemProvider.hasItemConformingToTypeIdentifier(UTType.text.identifier) {
+            saveText(itemProvider: itemProvider)
+        } else if itemProvider.hasItemConformingToTypeIdentifier(UTType.image.identifier) {
+            saveImage(itemProvider: itemProvider)
+        } else if itemProvider.hasItemConformingToTypeIdentifier(UTType.data.identifier) {
+            saveData(itemProvider: itemProvider)
+        } else {
+            finishImport()
+        }
+    }
+
+    func saveData(itemProvider: NSItemProvider) {
+        itemProvider.loadFileRepresentation(forTypeIdentifier: UTType.data.identifier) { [weak self] url, error in
+            guard error == nil else {
+                self?.finishImport()
+                return
+            }
+            if let url, let data = try? Data(contentsOf: url) {
+                self?.saveData(data, type: .data)
+            }
+            self?.finishImport()
+        }
+    }
+
+    func saveUrl(itemProvider: NSItemProvider) {
+        itemProvider.loadItem(forTypeIdentifier: UTType.url.identifier) { [weak self] item, error in
+            guard error == nil else {
+                self?.finishImport()
+                return
+            }
+            if let url = item as? URL, let data = url.absoluteString.data(using: .utf8) {
+                self?.saveData(data, type: UTType.url)
+            }
+            self?.finishImport()
+        }
+    }
+
+    func saveText(itemProvider: NSItemProvider) {
+        itemProvider.loadItem(forTypeIdentifier: UTType.text.identifier) { [weak self] item, error in
+            guard error == nil else {
+                self?.finishImport()
+                return
+            }
+            if let item = item as? String, let data = item.data(using: .utf8) {
+                self?.saveData(data, type: UTType.text)
+            }
+            self?.finishImport()
+        }
+    }
+
+    func saveImage(itemProvider: NSItemProvider) {
+        itemProvider.loadItem(forTypeIdentifier: UTType.image.identifier) { [weak self] item, error in
+            guard error == nil else {
+                self?.finishImport()
+                return
+            }
+            if let item = item as? UIImage, let pngData = item.pngData() {
+                self?.saveData(pngData, type: UTType.image)
+            } else if let url = item as? URL, let data = try? Data(contentsOf: url) {
+                self?.saveData(data, type: .image)
+            }
+            self?.finishImport()
+        }
+    }
+
+    func saveData(_ data: Data, type: UTType) {
+        let date = Date()
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "YYYY-MM-dd-HH-mm-ss"
```

Comment:
Send 3 photos from Photos -> they all got the same date -> You will only get 1 file